subsys/bluetooth/host/adv.c

```diff
@@ -1360,18 +1373,22 @@ int bt_le_ext_adv_delete(struct bt_le_ext_adv *adv)
 static void adv_timeout(struct k_work *work)
 {
 	int err = 0;
-#if defined(CONFIG_BT_EXT_ADV)
+
 	struct k_work_delayable *dwork;
 	struct bt_le_ext_adv *adv;
 
 	dwork = k_work_delayable_from_work(work);
 	adv = CONTAINER_OF(dwork, struct bt_le_ext_adv, timeout_work);
 
-	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
+#if defined(CONFIG_BT_EXT_ADV)
+	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV) &&
+		adv->timeout_work_active) {
 		err = bt_le_ext_adv_stop(adv);
 	}
 #else
-	err = bt_le_adv_stop();
+	if (adv->timeout_work_active) {
+		err = bt_le_adv_stop();
+	}
 #endif
 	__ASSERT(err != 0, "Limited Advertising timeout reached, "
```

Comment:
The assert seems like a bad idea. You'd have to be 100% sure that you don't have any race on who's stopping the advertiser.
And failing to stop and advertiser (because it is already stopped) isn't something that should warrant an assert in my opinion. I would feel much better if this was a BT_WARN instead.